index.html

```diff
@@ -57,10 +57,202 @@
+*mount* une nouvelle instance du nouveau composant.
+
+---
+# Algorithme de mise à jour: conséquences
+
+1. L'algorithme est joué à chaque update ⇨ attention à ne pas réaliser d'updates
+plus souvent que nécessaire.
+2. Les méthodes `render` ainsi que les autres méthodes du cycle de vie d'un
+rendu sont exécutées à chaque update ⇨ attention à leurs caractéristiques de
+performance.
+3. L'algorithme lui-même est O(n) d'après les développeurs de React. Donc un
+arbre plus gros va linéairement augmenter la durée de l'algorithme.
+
+---
+# 1. Ne pas réaliser plus d'updates que nécessaire
+
+Une update sera déclenchée à chaque appel à `setState`. Donc:
+
+* éviter de mettre dans le `state` des choses inutiles au `render`.
+* éviter de déclencher trop de `setState` en réponse à des event handlers.
+
+---
+# 2. Attention à la performance de `render`
+
+* traiter le rendu d'une liste: diviser en morceaux ou rendre la liste virtuelle.
+* attention aux calculs coûteux: il vaut mieux les faire en amont et stocker le
+résultat dans le state.
+* c'est la même chose pour `componentWillUpdate`, `componentDidUpdate` etc.
+* un `setState` dans `componentDidUpdate` va déclencher un nouveau cycle d'update
+
+---
+# 3. Aider l'algorithme de réconciliation à vous aider
+
+* essayer de changer l'état au plus proche du changement d'UI. Redux peut aider
+ici.
+* `shouldComponentUpdate` et `PureComponent` sont des bons outils pour "couper"
+des branches de l'arbre.
+* lorsqu'on rend des listes, il ne faut pas oublier de mettre l'attribut *key*
+
+---
+# Reparlons de `shouldComponentUpdate`
+
+* appelée au début du processus d'update d'un composant
+* si la méthode retourne `false` le rendu est arrêté pour ce composant... et
+tous ses enfants
+* très bonne manière de diminuer le coût d'une passe de rendu
+
+---
+# `shouldComponentUpdate`
+## Un exemple
+
+.smallest[
+```js
+class ComplexPanel extends React.Component {
+  // Note: this syntax, new but supported by Babel, automatically binds the
+  // method with the object instance.
+  onClick = () => {
+    this.setState({ detailsOpen: true });
+  }
+
+  // Return false to avoid a render
+* shouldComponentUpdate(nextProps, nextState) {
+    // Note: this works only if `summary` and `content` are primitive data
+    // (eg: string, number) or immutable data
+    // (keep reading to know more about this)
+*   return nextProps.summary !== this.props.summary
+*     || nextProps.content !== this.props.content
+*     || nextState.detailsOpen !== this.state.detailsOpen;
+  }
+
+  render() {
+    return (
+      <div>
+        <ComplexSummary summary={this.props.summary} onClick={this.onClick}/>
+        {this.state.detailsOpen
+          ? <ComplexContent content={this.props.content} />
+          : null}
+      </div>
+    );
+  }
+}
+```
+]
+
+---
+# `shouldComponentUpdate`
+## Les pièges
+
+Exécuté à chaque render cycle, donc il doit être performant.
+
+Donc pas de comparaison coûteuse.
```

Comment:
Est-il possible de considerer `!shallowEqual(this.props, nextProps) || !shallowEqual(this.state, nextState)`  comme un comparaison coûteuse? :D

Reply:
justement non, ça a un coût marginal, surtout par rapport au reste du traitement :)